dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/impl/TCPUtils.java

```diff
@@ -0,0 +1,124 @@
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.openliberty.netty.internal.BootstrapConfiguration;
+import io.openliberty.netty.internal.ChannelInitializerWrapper;
+import io.openliberty.netty.internal.ConfigConstants;
+import io.openliberty.netty.internal.ServerBootstrapExtended;
+import io.openliberty.netty.internal.exception.NettyException;
+import io.openliberty.netty.internal.tcp.TCPChannelInitializerImpl;
+import io.openliberty.netty.internal.tcp.TCPMessageConstants;
+import io.openliberty.netty.internal.tcp.TCPConfigurationImpl;
+
+public class TCPUtils {
+
+    private static final TraceComponent tc = Tr.register(NettyFrameworkImpl.class, TCPMessageConstants.NETTY_TRACE_NAME, TCPMessageConstants.TCP_BUNDLE);
+    private static final int timeBetweenRetriesMsec = 1000; // make this non-configurable
+
+    public static ServerBootstrapExtended createTCPBootstrap(NettyFrameworkImpl framework, Map<String, Object> tcpOptions) throws NettyException {
+        // TODO: toggle inbound
+        BootstrapConfiguration config = new TCPConfigurationImpl(tcpOptions, true);
+        ServerBootstrapExtended bs = new ServerBootstrapExtended();
+        bs.group(framework.parentGroup, framework.childGroup);
+        bs.channel(NioServerSocketChannel.class);
+        // apply the existing user config to the Netty TCP channel
+        bs.applyConfiguration(config);
+        ChannelInitializerWrapper tcpInitializer = new TCPChannelInitializerImpl(config);
+        bs.setBaseInitializer(tcpInitializer);
+        return bs;
+    }
+
+    private static ChannelFuture bindHelper(NettyFrameworkImpl framework, ServerBootstrapExtended bootstrap, String inetHost, int inetPort, ChannelFutureListener bindListener, final int retryCount) {
+        ChannelFuture bindFuture = bootstrap.bind(inetHost, inetPort);
+        if (bindListener != null) {
+            bindFuture.addListener(bindListener);
+        }
+        final String channelName = ((TCPConfigurationImpl) bootstrap.getConfiguration()).getExternalName();
+        
+        bindFuture.addListener(future -> {
+            if (future.isSuccess()) {
+                framework.activeChannels.add(bindFuture.channel());
+                
+                // set common channel attrs
+                bindFuture.channel().attr(ConfigConstants.NameKey).set(channelName);
+                bindFuture.channel().attr(ConfigConstants.HostKey).set(inetHost);
+                bindFuture.channel().attr(ConfigConstants.PortKey).set(inetPort);
+
+                Tr.info(tc, TCPMessageConstants.TCP_CHANNEL_STARTED, new Object[] { channelName, inetHost, String.valueOf(inetPort) });
+            } else {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "bindHelper failed due to: " + future.cause().getMessage());
```

Comment:
I think it would be a good idea to include the channelName, similarly to line 66 above?